java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessConnection.java

```diff
@@ -247,10 +249,8 @@ public void setReadOnly(boolean readOnly) throws SQLException {
                 Constants.SQLExceptionMessages.METHOD_CALLED_ON_OPEN_TRANSACTION);
         }
 
-        if (readOnly) {
-            throw new SQLFeatureNotSupportedException(Constants.SQLExceptionMessages.READ_ONLY);
-        }
-        this.readOnly = false;
+        // TODO implement read only properly
+        this.readOnly = readOnly;
```

Comment:
Fair enough. We have this on a private branch so it's not blocking to me but when I have a few spare cycles I'll implement this properly.